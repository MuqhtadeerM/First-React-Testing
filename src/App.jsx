import "./App.css";
import OnChnage from "./components/onChangeInputTest";
import Users from "./components/class-components/class";
import FunctionalComponents from "./components/functional-components/functional";
import GetByRole from "./components/RTL Query/getByRole/getby";
import GetAllRole from "./components/RTL Query/getAllEByRole/getAllby";
import GetLabel from "./components/RTL Query/getLabel/getLabel";
function App() {
  return (
    <div>
      {/* <p>First React Test Case</p>
      <p>Muqhtadeer</p>
      <img
        title="This is an AI Generated Image"
        src="https://media.istockphoto.com/id/517188688/photo/mountain-landscape.jpg?s=1024x1024&w=0&k=20&c=z8_rWaI8x4zApNEEG9DnWlGXyDIXe-OmsAyQ5fGPVV8="
      /> */}

      {/* <OnChange /> */}

      {/* Classs components */}
      <Users />

      {/* functional components */}
      <FunctionalComponents />

      {/* GetByRole usage */}
      <GetByRole />

      {/* getAllByRoles */}
      <GetAllRole />

      {/* getByLabelText */}
      <GetLabel />
    </div>
  );
}

export default App;
