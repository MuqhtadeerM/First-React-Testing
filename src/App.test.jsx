import { render, screen } from "@testing-library/react";
import { expect } from "vitest";
import App from "./App";

test("Test First React app case", () => {
  render(<App />);
  const text = screen.getByText(/First react Test Case/i);
  const title = screen.getByTitle(/this is an ai generated image/i);
  const text2 = screen.getByText(/Muqhtadeer/i);

  expect(text).toBeInTheDocument();
  expect(title).toBeInTheDocument();
  expect(text2).toBeInTheDocument();
});

test("Test First React app case2", () => {
  render(<App />);
  const text = screen.getByText(/First react Test Case/i);
  const title = screen.getByTitle(/this is an ai generated image/i);
  const text2 = screen.getByText(/Muqhtadeer/i);

  expect(text).toBeInTheDocument();
  expect(title).toBeInTheDocument();
  expect(text2).toBeInTheDocument();
});
