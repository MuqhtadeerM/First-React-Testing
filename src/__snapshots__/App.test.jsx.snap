// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot for app component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          First React Test Case
        </p>
        <p>
          Muqhtadeer
        </p>
        <img
          src="https://media.istockphoto.com/id/517188688/photo/mountain-landscape.jpg?s=1024x1024&w=0&k=20&c=z8_rWaI8x4zApNEEG9DnWlGXyDIXe-OmsAyQ5fGPVV8="
          title="This is an AI Generated Image"
        />
        <div>
          <h1>
            Test the On Change Event With Input Text
          </h1>
          <input
            type="text"
            value=""
          />
          <button>
            Update Data
          </button>
          <h1 />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        First React Test Case
      </p>
      <p>
        Muqhtadeer
      </p>
      <img
        src="https://media.istockphoto.com/id/517188688/photo/mountain-landscape.jpg?s=1024x1024&w=0&k=20&c=z8_rWaI8x4zApNEEG9DnWlGXyDIXe-OmsAyQ5fGPVV8="
        title="This is an AI Generated Image"
      />
      <div>
        <h1>
          Test the On Change Event With Input Text
        </h1>
        <input
          type="text"
          value=""
        />
        <button>
          Update Data
        </button>
        <h1 />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
